generate.py

```diff
@@ -167,139 +211,127 @@ def generate(args):
-            trips_above_min_soc += desired_soc > args.min_soc
-            desired_soc = max(args.min_soc, desired_soc)
-            # update initial desired SoC
-            v["desired_soc"] = v["desired_soc"] or desired_soc
+            v["last_arrival_idx"] = len(events["vehicle_events"])
 
             events["vehicle_events"].append({
-                "signal_time": arrival_time.isoformat(),
-                "start_time": arrival_time.isoformat(),
+                "signal_time": arrival.isoformat(),
+                "start_time": arrival.isoformat(),
                 "vehicle_id": v_id,
                 "event_type": "arrival",
                 "update": {
                     "connected_charging_station": "CS_" + v_id,
-                    "estimated_time_of_departure": next_dep_time.isoformat(),
-                    "desired_soc": desired_soc,
+                    "estimated_time_of_departure": None,
+                    "desired_soc": None,
                     "soc_delta": -soc_delta
                 }
             })
 
-    # reset initial SOC
+        # generate prices for the day
+        if not args.include_price_csv and now < stop:
+            morning = now + datetime.timedelta(hours=6)
+            evening_by_month = now + datetime.timedelta(hours=22-abs(6-now.month))
+            events['grid_operator_signals'] += [{
+                # day (6-evening): 15ct
+                "signal_time": max(start, now-daily).isoformat(),
+                "grid_connector_id": "GC1",
+                "start_time": morning.isoformat(),
+                "cost": {
+                    "type": "fixed",
+                    "value": 0.15 + random.gauss(0, 0.05)
+                }
+            }, {
+                # night (depending on month - 6): 5ct
+                "signal_time": max(start, now-daily).isoformat(),
+                "grid_connector_id": "GC1",
+                "start_time": evening_by_month.isoformat(),
+                "cost": {
+                    "type": "fixed",
+                    "value": 0.05 + random.gauss(0, 0.03)
+                }
+            }]
+
+    # end of scenario
+
+    # remove temporary information
     for v in vehicles.values():
-        del v["distance"]
-        del v["departure"]
+        del v["last_arrival_idx"]
 
     j = {
         "scenario": {
             "start_time": start.isoformat(),
-            "interval": int(interval.days * 24 * 60 + interval.seconds/60),
-            "n_intervals": int((stop - start) / interval)
+            # "stop_time": stop.isoformat(),
+            "interval": interval.days * 24 * 60 + interval.seconds // 60,
+            "n_intervals": (stop - start) // interval
         },
         "constants": {
             "vehicle_types": vehicle_types,
             "vehicles": vehicles,
             "grid_connectors": {
                 "GC1": {
-                    "max_power": 630
+                    "max_power": 630,
```

Comment:
Bitte für generate.py UND generate_from_download.py standardmäßig auf 530 kW setzen, aber auch als arg, sodass es in deden config-Dateien eingestellt werden kann